opsdroid/database/matrix/__init__.py

```diff
@@ -0,0 +1,250 @@
+
+        if self.should_migrate:
+            await self.migrate_database()
+
+        # If the single state key flag is set then use that else use state key.
+        state_key = (
+            "" if self._single_state_key is True else self._single_state_key or key
+        )
+
+        value = {key: value}
+
+        data = await self.get(key, get_full=True)
+        if data is None:
+            data = {}
+
+        if {**data, **value} == data:
+            _LOGGER.error("Not updating matrix state, as content hasn't changed.")
+            return
+        else:
+            data = {**data, **value}
+
+        if self.is_room_encrypted and self.should_encrypt:
+            room_event = await self.opsdroid.send(
+                GenericMatrixRoomEvent(
+                    target=self.room_id, content=value, event_type=self._event_type
+                )
+            )
+            data[key] = {"encrypted_val": room_event.event_id}
+
+        _LOGGER.debug(f"Putting {key} into matrix room {self.room_id} with {data}")
+
+        await self.opsdroid.send(
+            MatrixStateEvent(
+                self._event_type,
+                content=data,
+                target=self.room_id,
+                connector=self.connector,
+                state_key=state_key,
+            )
+        )
+
+        return True
+
+    async def get(self, key, get_full=False):
+        """Get a value from the database for a given key."""
+
+        if self.should_migrate:
+            await self.migrate_database()
+
+        # If the single state key flag is set then use that else use state key.
+        state_key = (
+            "" if self._single_state_key is True else self._single_state_key or key
+        )
+
+        _LOGGER.debug(
+            f"Getting {key} from matrix room {self.room_id} with state_key={state_key}."
+        )
+
+        ori_data = await self.connector.connection.room_get_state_event(
+            room_id=self.room_id, event_type=self._event_type, state_key=state_key,
+        )
+        if isinstance(ori_data, RoomGetStateEventError):
+            raise RuntimeError(
+                f"Error getting {key} from matrix room {self.room_id}: {ori_data.message}({ori_data.status_code})"
+            )
+        elif ori_data.transport_response.status == 404:
+            _LOGGER.error(
+                f"Error getting {key} from matrix room {self.room_id}: Event not found"
+            )
+            return
+
+        data = ori_data.content
+
+        _LOGGER.debug(f"Got {data} from state request.")
+
+        if self._single_state_key:
+            if key in data:
+                data = {key: data[key]}
+            elif not get_full:
+                _LOGGER.debug(f"{key} doesn't exist in database")
+                return
+            else:
+                return data
+
+        for k, v in data.items():
+            if isinstance(v, dict) and "encrypted_val" in v:
```

Comment:
We could be slightly stricter here and do:
```suggestion
            if isinstance(v, dict) and len(v) == 1 and "encrypted_val" in v:
```